src/ar.erl

```diff
@@ -151,67 +122,81 @@ main("") ->
 	),
 	erlang:halt();
 main(Args) ->
-	start(parse(Args)).
-
-parse(Args) -> parse(Args, #opts{}).
-parse([], O) -> O;
-parse(["init"|Rest], O) ->
-	parse(Rest, O#opts { init = true });
-parse(["mine"|Rest], O) ->
-	parse(Rest, O#opts { mine = true });
-parse(["peer", Peer|Rest], O = #opts { peers = Ps }) ->
-	parse(Rest, O#opts { peers = [ar_util:parse_peer(Peer)|Ps] });
-parse(["content_policy", File|Rest], O = #opts { content_policy_files = Files }) ->
-	parse(Rest, O#opts { content_policy_files = [File|Files] });
-parse(["transaction_blacklist", File|Rest], O = #opts { transaction_blacklist_files = Files } ) ->
-	parse(Rest, O#opts { transaction_blacklist_files = [File|Files] });
-parse(["port", Port|Rest], O) ->
-	parse(Rest, O#opts { port = list_to_integer(Port) });
-parse(["data_dir", DataDir|Rest], O) ->
-	parse(Rest, O#opts { data_dir = DataDir });
-parse(["diff", Diff|Rest], O) ->
-	parse(Rest, O#opts { diff = list_to_integer(Diff) });
-parse(["polling"|Rest], O) ->
-	parse(Rest, O#opts { polling = true });
-parse(["clean"|Rest], O) ->
-	parse(Rest, O#opts { clean = true });
-parse(["no_auto_join"|Rest], O) ->
-	parse(Rest, O#opts { auto_join = false });
-parse(["mining_addr", Addr|Rest], O) ->
-	parse(Rest, O#opts { mining_addr = ar_util:decode(Addr) });
-parse(["max_miners", Num|Rest], O) ->
-	parse(Rest, O#opts { max_miners = list_to_integer(Num) });
-parse(["new_mining_key"|Rest], O)->
-	parse(Rest, O#opts { new_key = true });
-parse(["disk_space", Size|Rest], O) ->
-	parse(Rest, O#opts { disk_space = (list_to_integer(Size)*1024*1024*1024) });
-parse(["load_mining_key", File|Rest], O)->
-	parse(Rest, O#opts { load_key = File });
-parse(["start_hash_list", BHLHash|Rest], O)->
-	parse(Rest, O#opts { start_hash_list = ar_util:decode(BHLHash) });
-parse(["benchmark"|Rest], O)->
-	parse(Rest, O#opts { benchmark = true });
-parse(["auto_update", "false" | Rest], O) ->
-	parse(Rest, O#opts { auto_update = false });
-parse(["auto_update", Addr | Rest], O) ->
-	parse(Rest, O#opts { auto_update = ar_util:decode(Addr) });
-parse(["internal_api_secret", Secret | Rest], O) when length(Secret) >= ?INTERNAL_API_SECRET_MIN_LEN ->
-	parse(Rest, O#opts { internal_api_secret = list_to_binary(Secret)});
-parse(["internal_api_secret", _ | _], _) ->
+	start(parse_config_file(Args, [], #opts{})).
+
+parse_config_file(["config_file", Path | Rest], Skipped, _) ->
+	{ok, Opts} = read_config_from_file(Path),
+	parse_config_file(Rest, Skipped, Opts);
+parse_config_file([Arg | Rest], Skipped, Opts) ->
+	parse_config_file(Rest, [Arg | Skipped], Opts);
+parse_config_file([], Skipped, Opts) ->
+	Args = lists:reverse(Skipped),
+	parse_cli_args(Args, Opts).
+
+read_config_from_file(Path) ->
+	case file:read_file(Path) of
+		{ok, FileData} -> ar_config:parse(FileData);
+		{error, _} -> {error, file_unreadable, Path}
+	end.
+
+parse_cli_args([], O) -> O;
```

Comment:
Since we have a new `ar_config` module (which is good thing), why don't we have this long function there? It would make sense to me.